api/observability/v1/input_types.go

```diff
@@ -0,0 +1,294 @@
+//
+// +kubebuilder:validation:Enum:=container;node
+type InfrastructureSource string
+
+const (
+	// InfrastructureSourceNode are journald logs from the node
+	InfrastructureSourceNode InfrastructureSource = "node"
+
+	// InfrastructureSourceContainer are container logs from workloads deployed
+	// in any of the following namespaces: default, kube*, openshift*
+	InfrastructureSourceContainer InfrastructureSource = "container"
+)
+
+var (
+	InfrastructureSources = []InfrastructureSource{
+		InfrastructureSourceNode,
+		InfrastructureSourceContainer,
+	}
+)
+
+// Infrastructure enables infrastructure logs.
+// Sources of these logs:
+// * container workloads deployed to namespaces: default, kube*, openshift*
+// * journald logs from cluster nodes
+type Infrastructure struct {
+	// Sources defines the list of infrastructure sources to collect.
+	// This field is optional and omission results in the collection of all infrastructure sources.
+	//
+	// +kubebuilder:validation:Optional
+	Sources []InfrastructureSource `json:"sources,omitempty"`
+}
+
+// AuditSource defines which type of audit log source is used.
+//
+// +kubebuilder:validation:Enum:=auditd;kubeAPI;openshiftAPI;ovn
+type AuditSource string
+
+const (
+	// AuditSourceKube are audit logs from kubernetes API servers
+	AuditSourceKube AuditSource = "kubeAPI"
+
+	// AuditSourceOpenShift are audit logs from OpenShift API servers
+	AuditSourceOpenShift AuditSource = "openshiftAPI"
+
+	// AuditSourceAuditd are audit logs from a node auditd service
+	AuditSourceAuditd AuditSource = "auditd"
+
+	// AuditSourceOVN are audit logs from an Open Virtual Network service
+	AuditSourceOVN AuditSource = "ovn"
+)
+
+var (
+	AuditSources = []AuditSource{
+		AuditSourceKube,
+		AuditSourceOpenShift,
+		AuditSourceAuditd,
+		AuditSourceOVN,
+	}
+)
+
+// Audit enables audit logs.
+type Audit struct {
+	// Sources defines the list of audit sources to collect.
+	// This field is optional and its exclusion results in the collection of all audit sources.
+	//
+	// +kubebuilder:validation:Optional
+	Sources []AuditSource `json:"sources,omitempty"`
+}
+
+// ReceiverType specifies the type of receiver that should be created.
+//
+// +kubebuilder:validation:Enum:=http;syslog
+type ReceiverType string
+
+const (
+	ReceiverTypeHttp   ReceiverType = "http"
+	ReceiverTypeSyslog ReceiverType = "syslog"
+)
+
+var (
+	ReceiverTypes = []ReceiverType{
+		ReceiverTypeHttp,
+		ReceiverTypeSyslog,
+	}
+)
+
+// ReceiverSpec is a union of input Receiver types.
+type ReceiverSpec struct {
+	// Type of Receiver plugin.
+	//
+	// +kubebuilder:validation:Required
+	Type ReceiverType `json:"type"`
+
+	// TLS contains settings for controlling options of TLS connections.
+	//
+	// +kubebuilder:validation:Optional
+	TLS *InputTLSSpec `json:"tls,omitempty"`
+
+	// Port the Receiver listens on. It must be a value between 1024 and 65535
+	//
+	// +kubebuilder:default:=8443
```

Comment:
Also use `+default:=8443` as it will apply the default to ApplyConfiguration should you ever generate that